plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -13,6 +13,81 @@ import { getByAge, UUIDT } from '../../utils/utils'
 
 type TeamCache<T> = Map<TeamId, [T, number]>
 
+function detectPropertyDefinitionTypes(isNumerical: boolean, value: any, key: string) {
+    let propertyType: PropertyType | null = null
+    let propertyTypeFormat: PropertyTypeFormat | null = null
+
+    /**
+     * Auto detecting unix timestamps is tricky. It's hard to know what is a big number or ID and what is a timestamp
+     *
+     * This tries to detect the most likely cases.
+     *
+     * * Numbers or Numeric Strings
+     * * That are either ten digits (seconds since unix epoch), or 13 digits (milliseconds since unix epoch),
+     * * or ten digits with numbers after the decimal place (whole seconds since unix epoch and fractions of a second)
+     * * where the property key includes either time or timestamp
+     *
+     * ten digits of seconds since epoch runs between Sep 09 2001 and Nov 20th 2286
+     *
+     * These are some representations from a variety of programming languages
+     *
+     * Python
+     * >>> datetime.now().timestamp()
+     * 1641477529.234715
+     *
+     * Ruby
+     * puts Time.now.to_i
+     * 1641477692
+     *
+     * Node JS
+     * console.log(Date.now())
+     * 1641477753371
+     *
+     * Java
+     * System.out.println(LocalDateTime.now().toEpochSecond(ZoneOffset.UTC));
+     * 1641478115
+     *
+     * SQL Lite
+     * select strftime('%s', 'now')
+     * 1641478347
+     *
+     * @param propertyValue
+     */
+    const detectUnixTimestamps = (propertyValue: string) => {
+        if (
+            (key.toLowerCase().indexOf('timestamp') > -1 || key.toLowerCase().indexOf('time') > -1) &&
+            propertyValue.match(/^(\d{10}|\d{13})(\.\d*)?$/)
+        ) {
+            propertyType = 'DateTime'
+            propertyTypeFormat = 'unix_timestamp'
+        }
+    }
+
+    if (isNumerical) {
+        propertyType = 'Numeric'
+
+        detectUnixTimestamps(value.toString())
+    }
+
+    if (typeof value === 'string') {
+        propertyType = 'String'
+
+        if (value.match(/^\d{4}-\d{2}-\d{2}$/)) {
+            propertyType = 'DateTime'
+            propertyTypeFormat = 'YYYY-MM-DD'
+        }
+
+        if (value.match(/^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}$/)) {
```

Comment:
ISO 8601 datetimes use a letter as the separator between date and time, most commonly "T", so something like this may cover more cases with a very similar risk of false positives:
```suggestion
        if (value.match(/^\d{4}-\d{2}-\d{2}[A-Za-z ]\d{2}:\d{2}:\d{2}$/)) {
```

Reply:
Will need (at some point) to update the property type formats to cover all the cases covered here https://clickhouse.com/docs/en/sql-reference/functions/type-conversion-functions/#parsedatetimebesteffort

(one of which is ISO-8601 and one this "non-standard" YYY-MM-DD hh:mm:ss")

It is probably not a large change to this PR to add all of those types (hopefully not famous last words)